src/components/resource/_fields/custom-embargo/custom-embargo-fields.js

```diff
@@ -32,124 +35,118 @@ function validateEmbargoValue(value) {
   return error;
 }
 
-function validateEmbargoUnit(value, { customEmbargoValue }) {
+function validateEmbargoUnit(embargoUnitValue, { customEmbargoPeriod }) {
   let error;
 
-  if (customEmbargoValue > 0 && !value) {
+  if (customEmbargoPeriod.length && customEmbargoPeriod[0].embargoValue > 0 && !embargoUnitValue) {
     error = <FormattedMessage id="ui-eholdings.validate.errors.embargoPeriod.unit" />;
   }
 
   return error;
 }
 
 export default class CustomEmbargoFields extends Component {
-  static propTypes = {
-    change: PropTypes.func.isRequired,
-    initial: PropTypes.object,
-    showInputs: PropTypes.bool
-  };
-
-  static defaultProps = {
-    initial: []
-  };
-
-  state = {
-    showInputs: this.props.showInputs
-  };
-
-  clearValues = () => {
-    this.props.change('customEmbargoValue', 0);
-    this.props.change('customEmbargoUnit', '');
-    this.toggleInputs();
-  }
+  renderInputs(fields, initial) {
+    const embargoValueField = (
+      <div
+        data-test-eholdings-custom-embargo-textfield
+        className={styles['custom-embargo-text-field']}
+      >
+        <FormattedMessage id="ui-eholdings.number">
+          {placeholder => (
+            <Field
+              name={`${fields.name}[0].embargoValue`}
+              component={TextField}
+              placeholder={placeholder}
+              autoFocus={!initial.length}
+              validate={validateEmbargoValue}
+            />
+          )}
+        </FormattedMessage>
+      </div>
+    );
 
-  toggleInputs = () => {
-    this.setState(({ showInputs }) => ({
-      showInputs: !showInputs
-    }));
-  }
+    const embargoUnitField = (
+      <div
+        data-test-eholdings-custom-embargo-select
+        className={styles['custom-embargo-select']}
+      >
+        <FormattedMessage
+          id="ui-eholdings.label.selectTimePeriod"
+        >
+          {placeholder => (
+            <Field
+              name={`${fields.name}[0].embargoUnit`}
+              component={Select}
+              validate={validateEmbargoUnit}
+              placeholder={placeholder}
+            >
+              <FormattedMessage id="ui-eholdings.label.days">
+                {(message) => <option value="Days">{message}</option>}
```

Comment:
Could we place FormattedMessage component inside option tag?

Reply:
> Could we place FormattedMessage component inside option tag?

It's not possible as it would lead to [object Object] rendered on the page